compiler/msgs.nim

```diff
@@ -486,6 +486,31 @@ proc formatMsg*(conf: ConfigRef; info: TLineInfo, msg: TMsgKind, arg: string): s
               else: ErrorTitle
   conf.toFileLineCol(info) & " " & title & getMessageStr(msg, arg)
 
+proc colorError(s: string, color: ForegroundColor): string =
+  template isQuote(val: untyped): untyped = val == '\''
+  template isNotQuote(val: untyped): untyped = val != '\''
+  var pos = 0
+  while pos < s.len:
+    if s[pos].isQuote:
+      inc pos
+      let start = pos
+      let nested = pos < s.high and s[pos].isQuote
+      inc pos
+      while (pos < s.len and s[pos].isNotQuote and not nested) or (nested and pos < s.high and (s[pos].isNotQuote or s[pos + 1].isNotQuote)):
```

Comment:
would be good to know when this logic breaks down, eg:
```nim
proc fn=
  let a = 0
  a = "will a sneaky quote in user code ' confuse this? "
```
and gracefully handle (eg just not colorize strings that aren't correctly parsed)